modules/drivers/canbus/sensor_canbus.h

```diff
@@ -185,6 +189,8 @@ Status SensorCanbus<SensorType>::Start() {
     const double duration = 1.0 / FLAGS_sensor_freq;
     timer_ = AdapterManager::CreateTimer(
         ros::Duration(duration), &SensorCanbus<SensorType>::OnTimer, this);
+  } else {
+    DataTrigger();
```

Comment:
Please correct me if I am wrong:, but DataTrigger seems to block the main thread, and "last step: publish monitor messages" will not be executed.

Reply:
Thank you for your suggestion!  It is fixed by creating a new thread for data trigger.